src/node.h

```diff
@@ -664,7 +664,18 @@ inline void NODE_SET_PROTOTYPE_METHOD(v8::Local<v8::FunctionTemplate> recv,
 #define NODE_SET_PROTOTYPE_METHOD node::NODE_SET_PROTOTYPE_METHOD
 
 // BINARY is a deprecated alias of LATIN1.
-enum encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER, LATIN1 = BINARY};
+// BASE64URL is not currently exposed to the JavaScript side.
+enum encoding {
+  ASCII,
+  UTF8,
+  BASE64,
+  UCS2,
+  BINARY,
+  HEX,
+  BUFFER,
+  BASE64URL,
+  LATIN1 = BINARY
+};
```

Comment:
@addaleax IIUC, this is OK ABI-wise because the new element doesn't change the other values of the enum? But `ParseEncoding` can now return this new value and programs that don't expect it would break? Can we do something about it (other than not backporting the feature)?

Reply:
@targos Yes, exactly, it’s ABI-compatible but not API-compatible, strictly speaking, and I don’t think we could really do something about it other than modifying `ParseEncoding()` to not return that value when called from userland code.

The problem with that is that, since we fall back to a default encoding in that case, `ParseEncoding(isolate, "base64url") == LATIN1`, which would seem like a bug.

I’d probably just leave this as-is, it’s unlikely to cause problems for a lot of addons (and even then only when somebody actually uses `base64url` with it).